soc/xtensa/intel_adsp/common/include/cpu_init.h

```diff
@@ -101,4 +101,42 @@ static ALWAYS_INLINE void cpu_early_init(void)
 	XTENSA_WSR("VECBASE", reg);
 }
 
+/* NOTE: This struct will grow with all values that have to be stored for
+ * proper cpu restore after PG.
+ */
+struct core_state {
+	uint32_t a0;
+	uint32_t a1;
+	uint32_t vecbase;
+	uint32_t excsave2;
+	uint32_t excsave3;
+	uint32_t thread_ptr;
+	uint32_t intenable;
+	uint32_t bctl;
```

Comment:
I can't see anything actually touching the intenable or bctl fields?  Did that get dropped or is it not part of the code motion?

Reply:
it is used by "core_desc[cpu].intenable = XTENSA_RSR("INTENABLE"); " and "	core_desc[cpu].bctl = DSPCS.bootctl[cpu].bctl;"